### ISSUE TITLE
Waiver Abuse

### ISSUE THREAD

**Issue by beaubreedlove** [beaubreedlove/FantasyFootball#6]
# tl;dr

I think Yahoo already does a good enough job at preventing "Waiver Abuse" to the point that we don't need any rules banning any such thing

# Background

Over the last few years there's always been mostly common agreement that Waiver Abuse is bad, but never a perfect way to define it. For this reason we've generally allowed people to veto _any_ action in the game so that all actions could be reviewed for abuse. I think it's time that we try to come to agreement about what waiver abuse is so we can determine whether any rules are needed to prevent it. This may also help free up the consciences of people who are hesitant to make moves that may look like waiver abuse, but aren't waiver abuse (see: [Handcuff Lottery](https://github.com/beaubreedlove/FantasyFootball/issues/7)).

I've done my best to sum up how I feel about a few scenarios:

## Good Moves:

1. Pick up player from WW or FA while dropping another player. This is a common and expected use of WW or FA.
2. Trade an even number of players. Example 1 for 1, 2 for 2, etc. This is a common and expected use of trades.
3. Pick up player from WW or FA without dropping a player. This feels like a fair move, though I'd argue that a manager should never allow their team to reach a state that makes this move possible.

## Bad Moves:

1. Before dropping player A to pick up player Z from FA, do the following: drop player A to WW while picking up player B from FA, drop player B to WW while picking up player C from FA, (repeat as long as you want), drop player Y to WW while picking up player Z from FA. Note that this results in players moving from FA to WW as a result of a single managers's actions and without any cost to that manager. This gives an unfair advantage because it effectively allows a team to choose to move players from FA to WW limitlessly without incurring any cost.

## Questionable Moves:

1. Drop a player to FA with out picking a player up. Reason it shouldn't be allowed: There's always a way to create waiver claims such that each player would have the same effect without the first player being dropped. Reason it should be allowed: This might require more waiver claims than is reasonable to set.
2. Make a trade that's unbalanced with respect to the number of players traded on each side. Reason it shouldn't be allowed: There's always a way to make a balanced trade by including the players that would have been immediately dropped by the team who's giving more players. Reason it should be allowed: Maybe the team trading less players shouldn't have to tell the other manager who they believe their bottom players are.

I could live with a set of rules that allowed for the questionable scenarios, but I'd rather not see them happen and I'd like to keep them on the table as possible evidence for collusion if a case ever arose. 

If we agree that the scenario I listed under Bad Moves is the only class of moves that we're trying to avoid, then I think we can get away with having no rules that mention Waiver Abuse. I say this because I recently became aware of [a rule that already exists on Yahoo](https://help.yahoo.com/kb/SLN6811.html)  which I will call the Yahoo Waiver Abuse Rule (YWAR) from now on. Due to our configuration of settings on Yahoo and the YWAR, when a player is dropped the same calendar day (PST) that they are picked up, they are dropped to FA instead of WW. This prevents the scenario described above under Bad Moves, because no more players are effectively moved to WW from FA than are allowed under common play. The most extreme scenario loop hole to this rule involves picking up a single player from FA while dropping a potentially desired player each day between Waivers clearing Tuesday and the first game Sunday morning. This combination of moves results in the maximum number of players in WW each day. I'm tempted to allow this tactic in the rules by not making a rule against it for two reasons:
1) The player who's committing "waiver abuse" is forced to defer picking up the play who they believe is best until 2 days before the game they play. This means they avoid any gain from having the most valuable player on their team. Other teams should notice this coming and pick up the best player.
2)  Dropping the worst player on a team for the best player on WW or FA multiple times per day has the same effect as dropping the dingle worst player for the single best player on WW or FA. Thus no negative effects are incurred by the other teams.

# If you disagree...

I can understand if not everyone agrees with my definition of Waiver abuse. I'd like to hear what you consider waiver abuse, if you disagree.

Regardless of your definition, I think a good way to prevent anything more that might more broadly be consider waiver abuse is to add some cost to making a move, preventing any advantage that making a high number of moves might give over other teams. If we don't find the Yahoo Waiver Abuse Rule sufficient for preventing Waiver Abuse, I would propose that we make a rule that all moves that effect the players on a team's roster cost $1. This $1 would be added to one ore more of the prize pools -- maybe we just give it all to 1st place.
